rust/cymbal/src/fingerprinting/v1.rs

```diff
@@ -0,0 +1,109 @@
+use crate::{
+    error::Error,
+    types::{frames::Frame, Exception},
+};
+use reqwest::Url;
+use sha2::{Digest, Sha256};
+
+// Given resolved Frames vector and the original Exception, we can now generate a fingerprint for it
+pub fn generate_fingerprint(
+    exception: &Exception,
+    resolved_frames: Vec<Frame>,
+) -> Result<String, Error> {
+    let mut fingerprint = format!(
+        "{}-{}",
+        exception.exception_type, exception.exception_message
+    );
+    for frame in resolved_frames {
```

Comment:
We should probably have some conditional logic here as to which frames we select for fingerprinting e.g.

1. Only use `in_app` frames if any exist, otherwise use the first frame
2. Only use `resolved` frames if any exist, otherwise use all frames (?)
3. Add a limit on the number of frames we include (maybe not if that feels a little arbitrary)